integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/mutation/CarbonProjectForUpdateCommand.scala

```diff
@@ -135,7 +136,22 @@ private[sql] case class CarbonProjectForUpdateCommand(
           else {
             Dataset.ofRows(sparkSession, plan)
           }
-
+          // If more than one value present for the update key, should fail the update
+          val ds = dataSet.select(CarbonCommonConstants.CARBON_IMPLICIT_COLUMN_TUPLEID)
```

Comment:
dataSet here includes all data which will be loaded, if the size of data which are to be updated is huge, I think this check will impact update performance.
It better check the count of updated data in 'CarbonIUDAnalysisRule.processUpdateQuery', before parse 'selectStmt' to 'selectPlan', for example:
line 115 in CarbonAnalysisRules.scala, check the count:
```
if (sparkSession.sql(selectStmt).count() > 1) {
  throw exception
}
```

Reply:
@jackylk @ravipesala what do you think about this?

Reply:
@zzcclp : I am filtering only tupleID column for this reason only, that too I am collecting only one row at driver. (collecting the count of tupleID which has more than one occurrence)

Reply:
yeah, I know that, but I still worry about the update performance when update huge data, even though just filter only tupleID column, but it still do 'group by' operation.
The way I mentioned above just needs to do a 'count' operation, and it can check this in earlier step.

Reply:
@zzcclp : I just modified like your solution and checked In "UpdateCarbonTableTestCase" itself 40 testcase failed. 

I will explain why it fails now, 

Here we don't want to throw exception if the `selectStmt` has more than one result, we need to throw exception when the it has more than one duplicate result. 

Example:
src has id = 1, 2. It need to update all the id, But the subquery results that 1,1,2. Now we don't know the 1 matches with the first or second. so only in this case we need to throw exception. 

Tell me if you have any other solution for this problem.

Reply:
OK, I got it. this way is ok to me, can you test some huge data to show how much this way will impact the update performance.
I will continue to try some way to check duplicated data on 'processUpdateQuery' step.